### PR TITLE
Graphics sockets to use UID folders

### LINKED ISSUE #1222: VirtualMachineInstance cannot launch when the VM name is too long
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**:

*1) VM name longer than 63 characters* 

VirtualMachineInstance with a VM name that is longer than 63 characters cannot launch.
Status of the vmi is empty and it has no events : 
```
./cluster/kubectl.sh describe vmi <long_vmi_name> 
```

```
 Status:
 Events:  <none>
```
Virt-Controller logs : 
```
level=info timestamp=2018-06-25T12:36:15.631975Z pos=vmi.go:157 
component=virt-controller service=http reason="invalid label value: 
\"bbb123456789012345678901234567890123456789012345678901234567890123\": 
must be no more than 63 characters" msg="reenqueuing VirtualMachineInstance 
default/bbb123456789012345678901234567890123456789012345678901234567890123"

```
*2) UNIX socket path is too long*

Even when the VM name is shorter than 63 characters, the UNIX socket path built from it can be too long and get rejected by the virt-handler.
UNIX socket path must be less than 108 bytes.

```
 Warning  SyncFailed          58s   virt-handler, node01       server error. command Launcher.Sync failed: 
virError(Code=1, Domain=10, Message='internal error: process exited while connecting to monitor: 
2018-06-25T11:11:03.026395Z qemu-system-x86_64: -chardev socket,id=charserial0,path=/var/run/kubevirt-
private/default/aaa123456789012345678901234567890123456789012345678901234567890/virt-
serial0,server,nowait: UNIX socket path '/var/run/kubevirt-private/default
/aaa123456789012345678901234567890123456789012345678901234567890/virt-serial0' is too long
Path must be less than 108 bytes')
```

**What you expected to happen**:
Successfully launch a VM when its name is too long.

**How to reproduce it (as minimally and precisely as possible)**:
1) Create a VM with a name that is longer than 63 characters.
2) Create a VM with a name that will create a UNIX socket path larger than 108 bytes.

### PR DESCRIPTION
**Pull request by yanirq**
**What this PR does / why we need it**:
Change Graphics socket path (console,VNC) to include UID folders instead of namespace/name folders.

**Which issue(s) this PR fixes** :
Handles #1222 

**Special notes for your reviewer**:
In case longer VM names will be allowed (longer than 63 characters) 
then the socket paths which currently are built according to :  
/var/run/kubevirt-private/**namespace**/**vm name**/virt-serial(or vnc)
will change to : 
/var/run/kubevirt-private/**vm UID**/virt-serial(or vnc)

This tackles the case when the vm name is very long. 

**Release note**:

```release-note
Ensure that socket paths for console/vnc are short enough to support long VMI names. 
This change will require already running VMs to be restarted after a KubeVirt upgrade to allow accessing the console or VNC endpoints again.
```

### CODE REVIEW

File: pkg/virt-launcher/virtwrap/api/converter_test.go [hunk 1]
```diff
@@ -321,7 +321,7 @@ var _ = Describe("Converter", func() {
       <model type="vga" heads="1" vram="16384"></model>
     </video>
     <graphics type="vnc">
-      <listen type="socket" socket="/var/run/kubevirt-private/mynamespace/testvmi/virt-vnc"></listen>
+      <listen type="socket" socket="/var/run/kubevirt-private/f4686d2c-6e8d-4335-b8fd-81bee22f4814/virt-vnc"></listen>
```
[the remark this answers is not in the document]
Comment (yanirq) [reply pkg/virt-launcher/virtwrap/api/converter_test.go]: Its taken from the above : vmi.ObjectMeta.UID = "f4686d2c-6e8d-4335-b8fd-81bee22f4814"
(see https://github.com/yanirq/kubevirt/blob/7faa676233fc63125b3b841a75440c2c73830421/pkg/virt-launcher/virtwrap/api/converter_test.go#L298)

**Comment by yanirq** [comment]
ok to test

**Comment by yanirq** [comment]
retest this please

**Comment by yanirq** [comment]
ci test please

**Comment by rmohr** [comment]
ok to test

**Comment by cynepco3hahue** [comment]
retest this please

**Comment by cynepco3hahue** [comment]
retest this please

**Comment by rmohr** [comment]
retest this please

**Comment by rmohr** [comment]
ok to test

**Comment by yanirq** [comment]
> Looks good so far. Can you add a test which verifies that a long name works? There is a good chance that we will still fail to start a vm because of too long paths for the socket which is used for the communication between virt-handler and virt-launcher.
@rmohr  I agree, but this test should be added to #1320 or add it after its merged.

**Pull request merged**